docker_cli/commands.py

```
@@ -0,0 +1,143 @@
+
+    image.drop_in()
+    print("Drop-in session ended.")
+    exit()
+
+
+def remove(
+    tags: Iterable[str],
+    force: bool = True,
+    quiet: bool = False,
+    ignore_prefix: bool = False,
+) -> None:
+    """
+    Remove all Docker images that fit a given tag or wildcard.
+
+    This tag or wildcard will take the form [UNIVERSAL PREFIX]-[tag or wildcard] if the
+    prefix does not already fit this.
+
+    Parameters
+    ----------
+    force : bool, optional
+        Whether or not to force the removal. Defaults to True.
+    tags : Iterable[str]
+        An iterable of tags or wildcards to be removed.
+    quiet : bool, optional
+        Whether or not to run quietly. Defaults to False.
+    ignore_prefix: bool, optional
+        Whether or not to ignore the universal prefix and only use the tag or wildcard.
+        Use with caution, as this will remove ALL images matching the wildcard.
+        e.g. remove(["*"], ignore_prefix = True) will remove all images.
+    """
+    force_arg = "--force " if force else ""
+    output = DEVNULL if quiet else None
+
+    # Search for and delete all images fitting each tag or wildcard.
+    for tag in tags:
+        prefix = universal_tag_prefix()
+        search = tag if (tag.startswith(prefix) or ignore_prefix) else f"{prefix}-{tag}"
+        if not quiet:
+            print(f"Attempting removal for tag: {search}")
+
+        # Search for all images whose name fits this tag, acquire a list
+        search_command = split(f'docker images --filter=reference="{search}" -q')
+        try:
+            search_result = run(search_command, text=True, stdout=PIPE, stderr=output)
+        # A CalledProcessError here indicates that the search failed. Skip to the next.
+        except CalledProcessError as err:
+            if not quiet:
+                print(
+                    f"Search for {search} failed with error code {err.returncode}. "
+                    "Proceeding."
+                )
+            continue
```

Hmm, why would we want to ignore errors resulting from the search command? It seems preferable to me to expose this exception to the user rather than catching it, printing the error code, and skipping on to the next search query.